contracts/services/osmosis-cl-withdrawer/src/msg.rs

```diff
@@ -0,0 +1,105 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::{Addr, Deps, DepsMut, Uint64};
+use cw_ownable::cw_ownable_query;
+
+use valence_macros::{valence_service_query, ValenceServiceInterface};
+use valence_osmosis_utils::utils::cl_utils::query_cl_pool;
+use valence_service_utils::{
+    error::ServiceError, msg::ServiceConfigValidation, ServiceAccountType,
+};
+
+#[cw_serde]
+pub enum ActionMsgs {
+    // liquidiate CL position by its id
+    WithdrawLiquidity {
+        position_id: Uint64,
+        liquidity_amount: String,
+    },
+}
+
+#[valence_service_query]
+#[cw_ownable_query]
+#[cw_serde]
+#[derive(QueryResponses)]
+pub enum QueryMsg {}
+
+#[cw_serde]
+#[derive(ValenceServiceInterface)]
+pub struct ServiceConfig {
+    pub input_addr: ServiceAccountType,
+    pub output_addr: ServiceAccountType,
+    pub pool_id: Uint64,
+}
+
+impl ServiceConfig {
+    pub fn new(
+        input_addr: impl Into<ServiceAccountType>,
+        output_addr: impl Into<ServiceAccountType>,
+        pool_id: Uint64,
+    ) -> Self {
+        ServiceConfig {
+            input_addr: input_addr.into(),
+            output_addr: output_addr.into(),
+            pool_id,
+        }
+    }
+
+    fn do_validate(
+        &self,
+        api: &dyn cosmwasm_std::Api,
+    ) -> Result<(Addr, Addr, Uint64), ServiceError> {
+        let input_addr = self.input_addr.to_addr(api)?;
+        let output_addr = self.output_addr.to_addr(api)?;
+
+        Ok((input_addr, output_addr, self.pool_id))
+    }
+}
+
+/// Validated service configuration
+#[cw_serde]
+pub struct Config {
+    pub input_addr: Addr,
+    pub output_addr: Addr,
+    pub pool_id: Uint64,
+}
+
+impl ServiceConfigValidation<Config> for ServiceConfig {
+    #[cfg(not(target_arch = "wasm32"))]
+    fn pre_validate(&self, api: &dyn cosmwasm_std::Api) -> Result<(), ServiceError> {
+        self.do_validate(api)?;
+        Ok(())
+    }
+
+    fn validate(&self, deps: Deps) -> Result<Config, ServiceError> {
+        let (input_addr, output_addr, pool_id) = self.do_validate(deps.api)?;
+        // sanity check
+        query_cl_pool(&deps, pool_id.u64())?;
+
+        Ok(Config {
+            input_addr,
+            output_addr,
+            pool_id,
+        })
+    }
+}
+
+impl ServiceConfigUpdate {
+    pub fn update_config(self, deps: DepsMut) -> Result<(), ServiceError> {
+        let mut config: Config = valence_service_base::load_config(deps.storage)?;
+
+        if let Some(input_addr) = self.input_addr {
+            config.input_addr = input_addr.to_addr(deps.api)?;
+        }
+
+        if let Some(output_addr) = self.output_addr {
+            config.output_addr = output_addr.to_addr(deps.api)?;
+        }
+
+        if let Some(pool) = self.pool_id {
+            query_cl_pool(&deps.as_ref(), pool.u64())?;
+            config.pool_id = pool;
+        }
+
+        Ok(())
+    }
```

Comment:
Don't forget to save the config here

Reply:
solid, this was missing in all osmo related services